lib/blob/blobstore.c

```diff
@@ -2777,6 +2777,37 @@ blob_calculate_lba_and_lba_count(struct spdk_blob *blob, uint64_t io_unit, uint6
 	}
 }
 
+static inline bool
+blob_ancestor_calc_lba_and_lba_count(struct spdk_blob *blob, uint64_t io_unit, uint64_t length,
+				     uint64_t *lba, uint64_t *lba_count)
+{
+	struct spdk_blob *orig_blob = blob;
+		
+	if (blob->parent_id == SPDK_BLOBID_INVALID) {
+		return false;
+	}
+	*lba_count = length;
+	while (blob->parent_id != SPDK_BLOBID_INVALID) {
+		spdk_blob_id blob_id = blob->parent_id;
+		blob = blob_lookup(blob->bs, blob_id);
+		if (blob == NULL) {
+			goto error;
+		} else {
+			if (!bs_io_unit_is_allocated(blob, io_unit)) {
+				continue;
+			}
+			*lba = bs_blob_io_unit_to_lba(blob, io_unit);
+			return true;
+		}
+	}
+	goto error;
```

Comment:
This `goto error` is redundant here, can be removed.

Reply:
right. Done fix.